src/Stryker.Core/Stryker.Core/Reporters/Html/Realtime/SseServer.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Sockets;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Stryker.Core.Reporters.Html.Realtime.Events;
+
+namespace Stryker.Core.Reporters.Html.Realtime;
+
+public class SseServer : ISseServer
+{
+    public int Port { get; set; }
+
+    private readonly HttpListener _listener;
+    private readonly List<StreamWriter> _writers;
+
+    public SseServer()
+    {
+        Port = FreeTcpPort();
+
+        _listener = new HttpListener();
+        _listener.Prefixes.Add($"http://localhost:{Port}/");
+        _writers = new List<StreamWriter>();
+    }
+
+    private static int FreeTcpPort()
+    {
+        var listener = new TcpListener(IPAddress.Loopback, 0);
+        listener.Start();
+        var port = ((IPEndPoint)listener.LocalEndpoint).Port;
+        listener.Stop();
+        return port;
+    }
+
+    public void OpenSseEndpoint()
+    {
+        _listener.Start();
+
+        Task.Run(ListenForConnectionsAsync);
+    }
+
+    private async Task ListenForConnectionsAsync()
+    {
+        while (true)
+        {
+            if (!_listener.IsListening)
+            {
+                return;
```

Comment:
This seems like an error condition. When does this happen and should we log this and/or throw an exception?

Reply:
I added this to satisfy sonarcloud, but I guess we could also put this condition in the `while (true) ...` loop

Reply:
Either this is something that can happen, and we should handle this gracefully (throwing or logging) so the user knows that the realtime reporting is not working, or this is a non-issue and should be marked as false positive in sonarcloud.

Reply:
I think it's a non-issue, but I like the condition to include `_listener.IsListening`, so it will stop doing the loop once the listener the realtime reporting is done.

Reply:
Okay, makes sense.